test/drivers/bench_cqrrp_gpu.cu

```diff
@@ -0,0 +1,273 @@
+        cudaStreamSynchronize(strm);
+        auto stop_orhr_col  = std::chrono::steady_clock::now();
+        auto diff_orhr_col  = std::chrono::duration_cast<std::chrono::milliseconds>(stop_orhr_col  - start_orhr_col).count();
+
+        // Mandatory data re-generation
+        data_regen(m_info, all_data, state);
+
+        // QRF part
+        auto start_qrf = std::chrono::steady_clock::now();
+        lapack::geqrf_work_size_bytes(m, n, all_data.A_device, m, &d_size_geqrf, &h_size_geqrf, lapack_queue);
+        d_work_geqrf = blas::device_malloc< char >( d_size_geqrf, lapack_queue );
+        std::vector<char> h_work_geqrf_vector( h_size_geqrf );
+        h_work_geqrf = h_work_geqrf_vector.data();
+        lapack::geqrf(m, n, all_data.A_device, m, all_data.tau_device, d_work_geqrf, d_size_geqrf, h_work_geqrf, h_size_geqrf, d_info, lapack_queue);
+        lapack_queue.sync();
+        auto stop_qrf  = std::chrono::steady_clock::now();
+        auto diff_qrf  = std::chrono::duration_cast<std::chrono::milliseconds>(stop_qrf  - start_qrf).count();
+        printf(" CholQR TIME (MS)   = %ld\n", diff_cholqr);
+        printf(" ORHR_COL TIME (MS) = %ld\n", diff_orhr_col);
+        printf(" QRF TIME (MS)      = %ld\n", diff_qrf);
+
+        std::ofstream file(output_filename, std::ios::app);
+        file << m << "  " << n << "  " << diff_cholqr << "  " << diff_orhr_col << "  " << diff_qrf << "\n";
+    }
+
+};
+/*
+TEST_P(BenchCQRRP, CQRRP_GPU_benchmark_16k) {
+    int64_t m            = std::pow(2, 14);
+    int64_t n            = std::pow(2, 14);
+    double d_factor      = 1.25;
+    int64_t b_sz         = GetParam();
+    double tol           = std::pow(std::numeric_limits<double>::epsilon(), 0.85);
+    auto state           = RandBLAS::RNGState();
+    bool profile_runtime = true;
+    bool run_qrf         = false;
+    if(b_sz == 120) {
+        run_qrf = true;
+    }
+
+    CQRRPBenchData<double> all_data(m, n);
+    RandLAPACK::gen::mat_gen_info<double> m_info(m, n, RandLAPACK::gen::gaussian);
+    RandLAPACK::gen::mat_gen<double, r123::Philox4x32>(m_info, all_data.A.data(), state);
+    cudaMemcpy(all_data.A_device, all_data.A.data(), m * n * sizeof(double), cudaMemcpyHostToDevice);
+
+
+    std::string file1 = "ICQRRP_GPU_runtime_breakdown_rows_"    + std::to_string(m)
+                                    + "_cols_"       + std::to_string(n)
+                                    + "_d_factor_"   + std::to_string(d_factor)
+                                    + ".dat";
+
+    std::string file2 = "ICQRRP_GPU_speed_rows_"    + std::to_string(m)
+                                    + "_cols_"       + std::to_string(n)
+                                    + "_d_factor_"   + std::to_string(d_factor)
+                                    + ".dat";
+
+    bench_CQRRP(profile_runtime, run_qrf, m_info, d_factor, tol, b_sz, all_data, state, file1, file2);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    CQRRP_GPU_16k_benchmarks,
```

Comment:
Now that this stuff is in another file I suggest you uncomment it. You can change the test/CMakeLists.txt so that the default run configuration skips test fixtures that start with "Bench".

Reply:
Not sure how to do it yet, but good idea.

Reply:
I'll probably open an issue on this.

Reply:
I checked options for how to achieve what you are suggesting and there are several potential ways of doing it.
I'll open an issue and we can discuss separately.